src/Scaffolding/VS.Web.CG.Mvc/Blazor/BlazorWebCRUDGenerator.cs

```diff
@@ -0,0 +1,352 @@
+            AppInfo = applicationInfo ?? throw new ArgumentNullException(nameof(applicationInfo)); ;
+            ModelTypesLocator = modelTypesLocator ?? throw new ArgumentNullException(nameof(modelTypesLocator));
+            FileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            FileLocator = fileLocator ?? throw new ArgumentNullException(nameof(fileLocator));
+            ProjectContext = projectContext ?? throw new ArgumentNullException(nameof(projectContext));
+            Workspace = workspace ?? throw new ArgumentNullException(nameof(workspace));
+            EntityFrameworkService = entityframeworkService ?? throw new ArgumentNullException(nameof(entityframeworkService));
+            ConsoleLogger = new ConsoleLogger(jsonOutput: false);
+        }
+
+        /// <summary>
+        /// Scaffold API Controller code into the provided (or created) Endpoints file. If no DbContext is provided, we will use the non-EF templates.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        public async Task GenerateCode(BlazorWebCRUDGeneratorCommandLineModel model)
+        {
+            model.ValidateCommandline();
+            var emptyTemplate = !string.IsNullOrEmpty(model.TemplateName) && model.TemplateName.Equals("empty", StringComparison.OrdinalIgnoreCase);
+            if (emptyTemplate)
+            {
+                ArgumentNullException.ThrowIfNull(model);
+                string emptyRazorName = "Empty";
+                var outputPath = ValidateAndGetOutputPath(string.Empty, emptyRazorName);
+
+                //arguments for 'dotnet new razorcomponent'
+                var additionalArgs = new List<string>()
+                {
+                    "razorcomponent",
+                    "--name",
+                    emptyRazorName,
+                    "--output",
+                    Path.GetDirectoryName(outputPath),
+                };
+
+                DotnetCommands.ExecuteDotnetNew(ProjectContext.ProjectFullPath, additionalArgs, Logger);
+                return;
+            }
+
+            //get model and dbcontext
+            var modelTypeAndContextModel = await ModelMetadataUtilities.GetModelEFMetadataBlazorAsync(
+                model,
+                EntityFrameworkService,
+                ModelTypesLocator,
+                Logger,
+                areaName: string.Empty);
+
+            //check if getting model and dbcontext was successfull
+            if (modelTypeAndContextModel is null ||
+                modelTypeAndContextModel.ContextProcessingResult is null ||
+                modelTypeAndContextModel.ContextProcessingResult.ContextProcessingStatus is ContextProcessingStatus.MissingContext ||
+                modelTypeAndContextModel.ModelType is null)
+            {
+                throw new InvalidOperationException("Unable to get model and/or dbcontext metadata.");
```

Comment:
We may want to define these user-facing messages in a separate file, with translation support. 

This could be implemented in our next revision of this component.

Reply:
We don't have localization for this repo but will move these to `MessageStrings.resx` later.